llvm/lib/Target/X86/X86InstrMisc.td

```diff
@@ -1244,6 +1244,15 @@ let Predicates = [HasBMI, HasEGPR], Defs = [EFLAGS] in {
   defm BLSI64 : bmi_bls<"blsi{q}", MRM3r, MRM3m, GR64, i64mem, WriteBLS, "_EVEX">, REX_W, EVEX;
 }
 
+let Predicates = [In64BitMode] in {
+  defm BLSR32 : bmi_bls<"blsr{l}", MRM1r, MRM1m, GR32, i32mem, WriteBLS, "_NF">, EVEX, EVEX_NF;
```

Comment:
Need to extract `VEX` from the class too. I made the change in https://github.com/llvm/llvm-project/commit/89ddd94516c880715ec78833f02a7af5e2766cb6. Need rebase.